Fix checking of %%MatrixMarket matrix coordinate pattern symmetric

GregorySchwing opened the pull request:
The %%MatrixMarket matrix coordinate pattern symmetric graphs don't have a weight column, which is accounted for in the helpers.h file.  However, a sanity test is conducted to evaluate whether the number of values parsed is equal to the number of edges times 3.  In the pattern graph, this produces an error since the number of "elements_per_line" for each edge is 2, not 3.  Therefore, I moved up the initialization of the "elements_per_line" variable and use this variable as opposed to a hardcoded 3 when checking sanity of the parser.

wheatman commented:
If you would like to add an example file can you just write up a small one that does not require not require citation for ease of use if anybody wants to use or download the code with different licenses?

GregorySchwing commented:
I handwrote a graph of three stars, connected by one edge to each star center.
![star(2)](https://user-images.githubusercontent.com/39970712/198155182-317d33dc-07cd-4691-ab01-8bf66b385ca0.png)

GregorySchwing committed:
add example file to try on linux

GregorySchwing committed:
fix issue when no weight column is present

wheatman commented on helpers.h:
```diff
@@ -316,8 +316,13 @@ get_edges_from_file_mtx(const std::string &filename, uint64_t *edge_count,
+  }
+
+  if (W.m - 8 != m * elements_per_line) {
+    std::cout << "Bad input file: length = " << W.m - 8 << elements_per_line << "m = " << elements_per_line * m
```
I think you want some commas between the output elements here, also make sure that the thing you are printing out is the same as the thing you say you are printing out.

GregorySchwing replied:
Added some commas and also print out the elements per line variable.  See example message below.

Bad input file: length = 39, elements per line 2, m = 40
%%MatrixMarket, matrix, coordinate, pattern, symmetric, 20, 20, 20, 2

GregorySchwing committed:
remove johnson graph

GregorySchwing committed:
add handwritten star graph

GregorySchwing committed:
fix an edge on star so center of stars form a triangle

GregorySchwing committed:
Updated error message
"Bad input file: length = 39, elements per line 2, m = 40
%%MatrixMarket, matrix, coordinate, pattern, symmetric, 20, 20, 20, 2"

wheatman commented:
Thanks so much for the fixes
